docs/source/concepts/ledger-model/ledger-integrity.rst

```diff
@@ -122,51 +249,18 @@ and inconsistent transactions.
    :align: center
    :width: 100%
 
-   The first two transactions violate the conditions of internally consistency.
+   The first two transactions violate the conditions of internal activeness consistency.
    The first transaction creates the `Iou` after exercising it consumingly, violating both conditions.
    The second transaction contains a (non-consuming) exercise on the `Iou` after a consuming one, violating the second condition.
    The last transaction is internally consistent.
 
-.. _def-contract-state:
-
-In addition to the consistency notions, the before-after relation on actions can also be used to define the notion of
-**contract state** at any point in a given transaction.
-The contract state is changed by creating the contract and by exercising it consumingly.
-At any point in a transaction, we can then define the latest state change in the obvious way.
-Then, given a point in a transaction, the contract state of `c` is:
-
-#. **active**, if the latest state change of `c` was a create;
-
-#. **archived**, if the latest state change of `c` was a consuming exercise was;
-
-#. **inexistent**, if `c` never changed state.
-
-A ledger is consistent for `c` exactly if **Exercise** and **Fetch** actions on `c` happen only when `c` is active, and **Create**
-actions only when `c` is inexistent.
-The figures below visualize the state of different contracts at all points in the example ledger.
-
-.. https://www.lucidchart.com/documents/edit/19226d95-e8ba-423a-8546-e5bae6bd3ab7
-.. figure:: ./images/consistency-paint-offer-activeness.svg
-   :align: center
-   :width: 100%
+Similar to input contracts, we define the input keys as the set that must be unassigned at the beginning of a transaction.
```

Comment:
They can also be unknown

Reply:
That's precisely what I meant with unassigned (=free or unknown). Added a definition.